templates/master/00-master/_base/files/usr-local-bin-openshift-recovery-tools-sh.yaml

```diff
@@ -34,19 +34,27 @@ contents:
       if [ -f "$ASSET_DIR/backup/etcd-ca-bundle.crt" ] && [ -f "$ASSET_DIR/backup/etcd-client.crt" ] && [ -f "$ASSET_DIR/backup/etcd-client.key" ]; then
          echo "etcd client certs already backed up and available $ASSET_DIR/backup/"
       else
-        for i in {1..10}; do
-            SECRET_DIR="${CONFIG_FILE_DIR}/static-pod-resources/kube-apiserver-pod-${i}/secrets/etcd-client"
-            CONFIGMAP_DIR="${CONFIG_FILE_DIR}/static-pod-resources/kube-apiserver-pod-${i}/configmaps/etcd-serving-ca"
-            if [ -f "$CONFIGMAP_DIR/ca-bundle.crt" ] && [ -f "$SECRET_DIR/tls.crt" ] && [ -f "$SECRET_DIR/tls.key" ]; then
-              cp $CONFIGMAP_DIR/ca-bundle.crt $ASSET_DIR/backup/etcd-ca-bundle.crt
-              cp $SECRET_DIR/tls.crt $ASSET_DIR/backup/etcd-client.crt
-              cp $SECRET_DIR/tls.key $ASSET_DIR/backup/etcd-client.key
-              break
-            else
-              echo "$SECRET_DIR does not contain etcd client certs, trying next source .."
-            fi
+        STATIC_DIRS=($(find "${CONFIG_FILE_DIR}/static-pod-resources" -maxdepth 1 -type d -name kube-apiserver-pod-*))
```

Comment:
A minor point: for finding the list of directories, using find is fine, but we could simply do ls as well.

```STATIC_DIRS=($(find "${CONFIG_FILE_DIR}/static-pod-resources" -maxdepth 1 -type d -name kube-apiserver-pod-*))```

can be:

```STATIC_DIRS=($(ls -d ${CONFIG_FILE_DIR}/static-pod-resources/kube-apiserver-pod-[0-9]*))```

But either should work.

Reply:
that is cleaner, updated thx.